pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -346,6 +346,39 @@ public synchronized TaskState getTaskState(String taskName) {
     return taskConfigs;
   }
 
+  /**
+   * Helper method to return a map of task names to corresponding task state
+   * where the task corresponds to the given Pinot table name. This is used to
+   * check status of all tasks for a given table.
+   * @param taskType Task Name
+   * @param tableNameWithType table name with type to filter on
+   * @return Map of filtered task name to corresponding state
+   */
+  public synchronized Map<String, TaskState> getTaskStatesByTable(String taskType, String tableNameWithType) {
+    Map<String, TaskState> filteredTaskStateMap = new HashMap<>();
+    Map<String, TaskState> taskStateMap = getTaskStates(taskType);
+
+    for (Map.Entry<String, TaskState> taskState : taskStateMap.entrySet()) {
+      String taskName = taskState.getKey();
+
+      // Iterate through all task configs associated with this task name
+      for (PinotTaskConfig taskConfig: getTaskConfigs(taskName)) {
+        Map<String, String> pinotConfigs = taskConfig.getConfigs();
+
+        // Filter task configs that matches this table name
+        if (pinotConfigs != null) {
+          String tableNameConfig = pinotConfigs.get("tableName");
```

Comment:
Can we have this string "tableName" defined in one place, so that when a table-specific task gets added, the table name key is always this string?

Reply:
sure thing. Lemme add that.

Reply:
@mcvsubbu there is a small issue. Looks like there's already a constant named 'TABLE_NAME' in CommonConstants:

```
    @Deprecated
    public static final String TABLE_NAME = "segment.table.name";
```

It'll be confusing if I add another of this. I think we should wait for this to be removed before adding the TABLE_NAME constant.